keadm/cmd/keadm/app/cmd/util/common_test.go

```diff
@@ -70,3 +81,139 @@ func TestManagedKubernetesVersion(t *testing.T) {
 		}
 	})
 }
+
+func TestPrivateDownloadServiceFile(t *testing.T) {
+	var (
+		componentType   types.ComponentType
+		targetVersion   semver.Version
+		serviceFilePath string
+	)
+	var check = func(serviceFilePath string) error {
+		_, err := os.Stat(serviceFilePath)
+		if err != nil {
+			return err
+		}
+		files, _ := ioutil.ReadDir(path.Dir(serviceFilePath))
+		if len(files) > 1 {
+			return fmt.Errorf("download redundancy files")
+		}
+		return err
+	}
+	var clean = func(testTmpDir string) {
+		dir, err := ioutil.ReadDir(testTmpDir)
+		if err != nil {
+			t.Fatalf("failed to clean test tmp dir!\n")
+		}
+		for _, d := range dir {
+			if err = os.RemoveAll(path.Join(testTmpDir, d.Name())); err != nil {
+				t.Fatalf("failed to clean test tmp dir!\n")
+			}
+		}
+	}
+
+	testTmpDir, err := ioutil.TempDir("", "kubeedge")
+	if err != nil {
+		t.Fatalf("failed to create temp dir for testing:{%s}\n", err.Error())
+	}
+
+	componentType = types.CloudCore
+	targetVersion = semver.Version{Major: 1, Minor: 0}
+	serviceFilePath = testTmpDir + "/" + CloudServiceFile
+	t.Run("test with downloading cloudcore service file if version < 1.1.0", func(t *testing.T) {
+		err := downloadServiceFile(componentType, targetVersion, testTmpDir)
+		if err != nil {
+			t.Fatalf("download should not return an error:{%s}\n", err.Error())
+		}
+		if err = check(serviceFilePath); !os.IsNotExist(err) {
+			if err == nil {
+				err = errors.New("nil")
+			}
+			t.Fatalf("check should return error{%s} but not,error:{%s}\n", os.ErrNotExist, err.Error())
+		}
+		clean(testTmpDir)
+	})
+
+	componentType = types.CloudCore
+	targetVersion = semver.Version{Major: 1, Minor: 1}
+	serviceFilePath = testTmpDir + "/" + CloudServiceFile
+	t.Run("test with downloading cloudcore service file if version = 1.1.0", func(t *testing.T) {
+		err := downloadServiceFile(componentType, targetVersion, testTmpDir)
+		if err != nil {
+			t.Fatalf("download should not return an error:{%s}\n", err.Error())
+		}
+		if err = check(serviceFilePath); !os.IsNotExist(err) {
+			if err == nil {
+				err = errors.New("nil")
+			}
+			t.Fatalf("check should return error{%s} but not,error:{%s}\n", os.ErrNotExist, err.Error())
+		}
+		clean(testTmpDir)
+	})
+
+	componentType = types.CloudCore
+	targetVersion, _ = semver.Make(types.LatestKubeEdgeVersion)
```

Comment:
Please use the `DefaultKubeEdgeVersion` here, no need create a new one, `DefaultKubeEdgeVersion` is the latest one.

Reply:
Done : )